bigquery/bigquery_analyticshub_quickstart/main.go

```diff
@@ -0,0 +1,167 @@
+	// print information about it.
+	exchange, err := createOrGetDataExchange(ctx, dataExchClient, *projectID, *location, *exchangeID)
+	if err != nil {
+		log.Fatalf("failed to get information about the exchange: %v", err)
+	}
+	fmt.Printf("\nData Exchange Information\n")
+	fmt.Printf("Exchange Name: %s\n", exchange.GetName())
+	if desc := exchange.GetDescription(); desc != "" {
+		fmt.Printf("Exchange Description: %s", desc)
+	}
+
+	// Finally, create a listing within the data exchange and print information about it.
+	listing, err := createListing(ctx, dataExchClient, *projectID, *location, *exchangeID, *listingID, *exampleDatasetSource)
+	if err != nil {
+		log.Fatalf("failed to create the listing within the exchange: %v", err)
+	}
+	fmt.Printf("\n\nListing Information\n")
+	fmt.Printf("Listing Name: %s\n", listing.GetName())
+	if desc := listing.GetDescription(); desc != "" {
+		fmt.Printf("Listing Description: %s\n", desc)
+	}
+	fmt.Printf("Listing State: %s\n", listing.GetState().String())
+	if source := listing.GetSource(); source != nil {
+		if dsSource, ok := source.(*dataexchangepb.Listing_BigqueryDataset); ok {
+			if dataset := dsSource.BigqueryDataset.GetDataset(); dataset != "" {
+				fmt.Printf("Source is a bigquery dataset: %s", dataset)
+			}
+		}
+	}
+	// Optionally, delete the data exchange at the end of the quickstart to clean up the resources used.
+	if *delete {
+		fmt.Printf("\n\n")
+		if err := deleteDataExchange(ctx, dataExchClient, *projectID, *location, *exchangeID); err != nil {
+			log.Fatalf("failed to delete exchange: %v", err)
+		}
+		fmt.Printf("Exchange projects/%s/locations/%s/dataExchanges/%s was deleted.\n", *projectID, *location, *exchangeID)
+	}
+	fmt.Printf("\nQuickstart completed.\n")
+}
+
+// createOrGetDataExchange creates an example data exchange, or returns information about the exchange already bearing
+// the example identifier.
+func createOrGetDataExchange(ctx context.Context, client *dataexchange.AnalyticsHubClient, projectID, location, exchangeID string) (*dataexchangepb.DataExchange, error) {
+	req := &dataexchangepb.CreateDataExchangeRequest{
+		Parent:         fmt.Sprintf("projects/%s/locations/%s", projectID, location),
+		DataExchangeId: exchangeID,
+		DataExchange: &dataexchangepb.DataExchange{
+			DisplayName:    "Example Data Exchange",
+			Description:    "Exchange created as part of an API quickstart",
+			PrimaryContact: "",
+			Documentation:  "https://link.to.optional.documentation/",
+		},
+	}
+
+	resp, err := client.CreateDataExchange(ctx, req)
+	if err == nil {
+		// With no error, return the provided response from a successful creation.
+		return resp, nil
+	}
+	// We'll handle one specific error case specially, the case of the exchange already existing.  In this instance,
+	// we'll issue a second request to fetch the exchange information for the already present exchange and return it.
+	if code := status.Code(err); code == codes.AlreadyExists {
```

Comment:
nit: Even though the error case is more invovled I would keep the happy path left justified and check if err != nil for consistency.